From foo@baz Wed Jul 29 15:38:00 PDT 2009
Date: Wed, 29 Jul 2009 15:38:00 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: remove custom rdmsrl and wrmsrl functions

From: Greg Kroah-Hartman <gregkh@suse.de>

Use the ones that the kernel provides, they do it correctly.

Cc: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/Hv.h          |    4 ++--
 drivers/staging/hv/include/osd.h |   35 -----------------------------------
 2 files changed, 2 insertions(+), 37 deletions(-)

--- a/drivers/staging/hv/Hv.h
+++ b/drivers/staging/hv/Hv.h
@@ -133,14 +133,14 @@ static inline unsigned long long ReadMsr
 {
 	unsigned long long val;
 
-	RDMSR(msr, val);
+	rdmsrl(msr, val);
 
 	return val;
 }
 
 static inline void WriteMsr(int msr, u64 val)
 {
-	WRMSR(msr, val);
+	wrmsrl(msr, val);
 
 	return;
 }
--- a/drivers/staging/hv/include/osd.h
+++ b/drivers/staging/hv/include/osd.h
@@ -66,41 +66,6 @@ struct osd_timer {
 };
 
 
-
-#ifdef __x86_64__
-
-#define RDMSR(reg, v) {                                                        \
-    u32 h, l;                                                                 \
-     __asm__ __volatile__("rdmsr"                                                               \
-    : "=a" (l), "=d" (h)                                                       \
-    : "c" (reg));                                                              \
-    v = (((u64)h) << 32) | l;                                                         \
-}
-
-#define WRMSR(reg, v) {                                                        \
-    u32 h, l;                                                               \
-    l = (u32)(((u64)(v)) & 0xFFFFFFFF);                                  \
-    h = (u32)((((u64)(v)) >> 32) & 0xFFFFFFFF);                          \
-     __asm__ __volatile__("wrmsr"                                              \
-    : /* no outputs */                                                         \
-    : "c" (reg), "a" (l), "d" (h));                                            \
-}
-
-#else
-
-#define RDMSR(reg, v) 			                                               \
-     __asm__ __volatile__("rdmsr" 	                                           \
-    : "=A" (v) 			                                                       \
-    : "c" (reg))
-
-#define WRMSR(reg, v) 			                                               \
-     __asm__ __volatile__("wrmsr" 	                                           \
-    : /* no outputs */ 				                                           \
-    : "c" (reg), "A" ((u64)v))
-
-#endif
-
-
 /* Osd routines */
 
 extern void* VirtualAllocExec(unsigned int size);
